repo/gen/crypto_generator.go

```diff
@@ -0,0 +1,53 @@
+package gen
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+
+	"github.com/libp2p/go-libp2p-crypto"
+	"github.com/libp2p/go-libp2p-peer"
+	"github.com/qri-io/doggos"
+)
+
+// CryptoGenerator is an interface for generating cryptographic info like private keys and peerIDs
+type CryptoGenerator interface {
+	GeneratePrivateKeyAndPeerID() (string, string)
```

Comment:
Can we get a comment here that indicates the returned private key value must be a base64-encoded private key string? In the future I'd love a test that checks a given implementation for such a thing, but we don't have time for such a thing atm